lib/kafka/produce_operation.rb

```diff
@@ -137,6 +137,9 @@ def handle_response(response)
               delay: ack_time - message.create_time,
             })
           end
+        rescue Kafka::MessageSizeTooLarge
+          @logger.error "Message is too large when writing to #{topic}/#{partition}"
+          @buffer.clear_messages(topic: topic, partition: partition)
```

Comment:
My rationale to _clear the buffer_ was to follow the same pattern as if we'd had no errors (see below in https://github.com/jasantunes/ruby-kafka/blob/b9745a7e1a3080c225e91ebf09cbedfba7c98af1/lib/kafka/produce_operation.rb#L164).

If `@buffer.clear_messages` is clearing unsent messages from the queue, then maybe we shouldn't be clearing the _entire_ buffer on no errors either, right?

Is there a way to clear only this particular message?